tests/test_pretest.py

```diff
@@ -152,6 +152,35 @@ def test_update_testbed_metadata(duthosts, tbinfo, fanouthosts):
     prepare_autonegtest_params(duthosts, fanouthosts)
 
 
+def test_update_snappi_testbed_metadata(duthosts, tbinfo, request):
+    """
+    Prepare metadata json for snappi tests, will be stored in metadata/snappi_tests/<tb>.json
+    """
+    pytest_require("tgen" in request.config.getoption("--topology"),
```

Comment:
Hi @amitpawar12 , lets do this so that it will not throw error if we don't provide "topology" cli option
```suggestion
     topology_list = request.config.getoption("--topology")
    pytest_require(topology_list is not None and "tgen" in topology_list, 
                   "Skip snappi metadata generation for non-tgen testbed")
```